stdlib/mexpr/pprint.mc

```diff
@@ -150,16 +154,41 @@ lang FunPrettyPrint = FunAst
 end
 
 lang RecordPrettyPrint = RecordAst
+  sem record2tuple =
+  | TmRecord t ->
+    match t.bindings with [] then
+      None ()
+    else
+      let keys = assocKeys {eq=eqstr} t.bindings in
```

Comment:
Does `assocKeys` guarantee that it produces a list with no duplicates? I'm guessing it depends on whether `AssocMap` was treated as an opaque datastructure and only `assoc*` functions were used, or if it was treated as on associative list.

Reply:
I assumed here that it's treated as an opaque datastructure, however, I don't know if we have a convention for that. In `pprintCode` for `TmRecord` we call `mapAccumL` directly on the associative list, so we are not consistent. Discuss during meeting perhaps?

Reply:
Take a look at `assocMapAccum` that I added previously. Maybe it makes sense to use it here?

Reply:
Thanks Daniel -- `assocMapAccum` did the trick of replacing `mapAccumL`, with the cost of an extra call to `assocValues` as `assocMapAccum` returns a new map. (I can revert this if we don't want this change.)